UnityProject/Assets/Scripts/Core/Random13.cs

```diff
@@ -0,0 +1,113 @@
+			135, 106, 63, 197, 195, 86, 96, 203, 113, 101, 170, 247, 181, 113,
+			80, 250, 108, 7, 255, 237, 129, 226, 79, 107, 112, 166, 103, 241,
+			24, 223, 239, 120, 198, 58, 60, 82, 128, 3, 184, 66, 143, 224,
+			145, 224, 81, 206, 163, 45, 63, 90, 168, 114, 59, 33, 159, 95,
+			28, 139, 123, 98, 125, 196, 15, 70, 194, 253, 54, 14, 109, 226,
+			71, 17, 161, 93, 186, 87, 244, 138, 20, 52, 123, 251, 26, 36,
+			17, 46, 52, 231, 232, 76, 31, 221, 84, 37, 216, 165, 212, 106,
+			197, 242, 98, 43, 39, 175, 254, 145, 190, 84, 118, 222, 187, 136,
+			120, 163, 236, 249
+		};
+
+		public int CurrentRandomTableIndex { get; private set; } = 0;
+		public int RandomNumber => randomTable[CurrentRandomTableIndex];
+		public Action<int> OnRandomTableIndexChanged;
+
+		private void ProgressIndex()
+		{
+			CurrentRandomTableIndex++;
+			if (CurrentRandomTableIndex >= randomTable.Length - 1) CurrentRandomTableIndex = 0;
+			OnRandomTableIndexChanged?.Invoke(randomTable[CurrentRandomTableIndex]);
+		}
+
+		/// <summary>
+		/// Quick way to randomly decide if something is true or false. Perfect for deterministic behavior as it relies on the
+		/// random table and its current used index.
+		/// </summary>
+		public static bool Prob()
+		{
+			if (Instance == null)
+			{
+				throw new Exception("Random13 was never created and stored anywhere.");
+			}
+			Instance.ProgressIndex();
+			return Instance.RandomNumber % 2 == 0;
+		}
+
+		/// <summary>
+		/// Consistent random numbers for things like damage.
+		/// </summary>
+		public static float RandomNumberWave(float baseNumber)
+		{
+			if (Instance == null)
+			{
+				throw new Exception("Random13 was never created and stored anywhere.");
+			}
+			Instance.ProgressIndex();
+			//Random number between 1 and 8 then multiplied by base damage.
+			return (Instance.RandomNumber % 8 + 1) * baseNumber;
+		}
+
+		/// <summary>
+		/// Random deterministic chance. Try your luck!
+		/// </summary>
+		/// <param name="chance">Chance between 0 and 1 (0.5 is 50%, etc.)</param>
+		public static bool ProbChance(double chance)
+		{
+			if (Instance == null)
+			{
+				throw new Exception("Random13 was never created and stored anywhere.");
+			}
+
+			if (chance is > 1 or < 0)
+			{
+				throw new ArgumentException("Chance must be between 0 and 1 as a double!!");
+			}
+			Instance.ProgressIndex();
+			var random = new Random(Instance.RandomNumber); //uses the table as a seed
+			return random.NextDouble() < chance;
+		}
+
+		/// <summary>
+		/// Uses a given time to determine the probability of having a 50/50 chance for something to happen.
+		/// Extremely fast and much more predictable.
+		/// </summary>
+		/// <param name="time">Usually DateTime.Now but a specific time can be given to replay expected events.</param>
+		/// <returns>50/50 chance for true or false based on current time given.</returns>
+		public static bool ProbFromTime(DateTime time)
+		{
+			if (time.Ticks % 2 == 0) return true;
+			return false;
+		}
+	}
```

Comment:
Why reinvent the wheel with random numbers? There's a perfectly fine stdlib class.
```suggestion
	/// <summary>
	/// Deterministic class for randomness.
	/// Best reserved for server logic, while client logic can use DMMath.
	/// </summary>
	public class Random13 : Random
	{
		// Use a specific seed for determinism (could also be unique per round)
		public Random13() : base(13)
		{
			Instance = this;
		}

		public static Random13 Instance;

		/// <summary>
		/// Randomly decide if something is true or false.
		/// </summary>
		public static bool Prob()
		{
			if (Instance == null)
			{
				throw new Exception("Random13 was never created and stored anywhere.");
			}

			return Instance.Next(2) == 0;
		}

		/// <summary>
		/// Consistent random numbers for things like damage.
		/// </summary>
		public static float RandomNumberWave(float baseNumber)
		{
			if (Instance == null)
			{
				throw new Exception("Random13 was never created and stored anywhere.");
			}

			//Random number between 1 and 8 then multiplied by base damage.
			return (Instance.Next(8) + 1) * baseNumber;
		}

		/// <summary>
		/// Random deterministic chance. Try your luck!
		/// </summary>
		/// <param name="chance">Chance between 0 and 1 (0.5 is 50%, etc.)</param>
		public static bool ProbChance(double chance)
		{
			if (Instance == null)
			{
				throw new Exception("Random13 was never created and stored anywhere.");
			}

			if (chance is > 1 or < 0)
			{
				throw new ArgumentException("Chance must be between 0 and 1 as a double!!");
			}
			return Instance.NextDouble() < chance;
		}

		/// <summary>
		/// Uses a given time to determine the probability of having a 50/50 chance for something to happen.
		/// Extremely fast and much more predictable.
		/// </summary>
		/// <param name="time">Usually DateTime.Now but a specific time can be given to replay expected events.</param>
		/// <returns>50/50 chance for true or false based on current time given.</returns>
		public static bool ProbFromTime(DateTime time)
		{
			if (time.Ticks % 2 == 0) return true;
			return false;
		}
	}
```

Reply:
`System.Random` doesn't guarantee stability across runtimes/versions. I'd recommend you google an algorithm online (they're really simple) instead, so you know it's consistent.